# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../unionfs/linux-2.6.34.1.diff
# Copyright (C) 2010 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

--- ./patch.vanilla	2010-08-06 12:15:30.515465855 +0000
+++ ./patch	2010-08-06 12:16:43.478344905 +0000
@@ -636,12 +636,14 @@
  		if (unlikely(ret <= 0))
  			goto out_release;
  
-@@ -1231,7 +1233,7 @@ static int direct_splice_actor(struct pipe_inode_info *pipe,
+@@ -1231,8 +1233,8 @@ static int direct_splice_actor(struct pipe_inode_info *pipe,
  {
  	struct file *file = sd->u.file;
  
--	return do_splice_from(pipe, file, &sd->pos, sd->total_len, sd->flags);
-+	return vfs_splice_from(pipe, file, &sd->pos, sd->total_len, sd->flags);
+-	return do_splice_from(pipe, file, &file->f_pos, sd->total_len,
+-			      sd->flags);
++	return vfs_splice_from(pipe, file, &file->f_pos, sd->total_len,
++			       sd->flags);
  }
  
  /**
